invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -78,6 +78,89 @@ def get_form_pids_config():
     return pids_providers
 
 
+class ExternalResourcesParser:
+    """Entry class for parsing records into their appropriate UI components."""
+
+    def __init__(self, resources):
+        """Constructor."""
+        self.resources = resources
+
+    def _parse_external_resources(self):
+        """Entry parser function for serializing resources based on their scheme."""
+        resources = []
+
+        for resource in self.resources:
+            scheme = resource["scheme"]
```

Comment:
Does the `scheme` allows providing a subclass of `ResourceParser` to override the default? I am not sure that I understand how it can be used :)/

Reply:
This is also in the above answer but I'll link an image here for clarity

![image](https://user-images.githubusercontent.com/59311729/225712431-745b3854-ab26-41d5-b76e-6bab7716f056.png)

The idea is that we would be able to parse not only `url` but the others too if we want (thus avoiding a bunch of ugly if chaining)
In retrospect this may be a bit overkill :sweat_smile:

Reply:
I would delegate all the complexity/freedom of generating the entry to these functions (or classes). e.g. if someone wants to render an entry for an EAN13 related identifier, it's completely up to them how they do it, and we don't do any validation or checks.